### PR TITLE
refactor mTLS documentation

### PR DESCRIPTION
**Pull request by inconshreveable**
This commit refactors the mTLS documentation to improve it with:

- Example usage in each of the SDKs, the Agent CLI, config file, Kubernetes Ingress controller and Edges.
- Upstream header behavior
- Validation of certificate basic constraints
- Usage and behavior with multiple CAs
- Error condition behaviors
- Which plans it is available on
- A simple example of how to try it out with curl
- The fields that are populated in the http_request_complete.v0 event when mTLS is enabled

### CODE REVIEW

File: docs/cloud-edge/modules/mutual-tls.md [hunk 1]
```diff
@@ -1,6 +1,204 @@
...
+:::
+
+You can [+1 the GitHub
+issue](https://github.com/ngrok/kubernetes-ingress-controller/issues/133)
+tracking its implementation.
+
+### Edges
+
+Mutual TLS is a supported Edge module. When the Mutual TLS module is configured
+via an Edge, you will specify one or more references to Certificate Authority
```
Comment (nijikokun) [inline docs/cloud-edge/modules/mutual-tls.md:104]:
```suggestion
via an Edge, you must specify one or more references to Certificate Authority
```

File: docs/cloud-edge/modules/mutual-tls.md [hunk 2]
```diff
@@ -1,6 +1,204 @@
 # Mutual TLS
 ----------------
 
-Also known as "TLS client authentication", connections must complete a mutual TLS handshake in which the client presents a valid certificate signed by any of the root certificate authorities that you upload.
+## Overview
 
-Note: Mutual TLS applies to all routes in an HTTPS edge.
+This module performs mutual TLS (mTLS) authentication when the ngrok edge terminates
+TLS on incoming connections to your HTTP endpoint. The client must present a
+valid TLS certificate that is signed by one of the specified CAs or the
```
Comment (nijikokun) [inline docs/cloud-edge/modules/mutual-tls.md:8]: Should we link specified CAs to the multiple CAs section below?

**Review by nijikokun — CHANGES_REQUESTED** [review]
Great stuff once again! Nice call out to the k8s issue 👍

Just some small edits here and one question.

File: docs/cloud-edge/modules/mutual-tls.md (hunk 2 above)
Comment (inconshreveable) [reply docs/cloud-edge/modules/mutual-tls.md:8]: i don't think so. it's unlikely the thing you're interested in when reading this section. using multiple CAs is relatively uncommon

File: docs/cloud-edge/modules/mutual-tls.md [hunk 3]
```diff
@@ -1,6 +1,204 @@
...
+5246](https://datatracker.ietf.org/doc/html/rfc5246#section-7.2). The most
+common alert code returned for a failed mutual TLS handshake is code 42
+(`bad_certificate`) which most TLS implementations will report with the error
+string string "bad certificate".
+
+## Try it out
+
+This example assumes you have an x509 private key and certificate encoded as
+PEM files called `client-key.pem` and `client-cert.pem`, respectively. The
+certificate must be signed by one of the CA certificates you provied to the
```
Comment (nijikokun) [inline docs/cloud-edge/modules/mutual-tls.md:185]:
```suggestion
certificate must be signed by one of the CA certificates you provided to the
```

**Review by nijikokun — APPROVED** [review]
LGTM!

**Pull request merged**